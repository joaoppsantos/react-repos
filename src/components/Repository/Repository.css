.fa-star {
    color: gold;
}

th {
    width: 20vh;
    height: 2.5vh;
}

p {
    font-size: 40px;
    padding-left: 3vh;
}
