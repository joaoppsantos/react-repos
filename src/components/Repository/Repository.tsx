import './Repository.css';

export interface RepositoryProps {
    forks: number;
    stars: number;
    url: string;
    name: string;
}

export const Repository = ({ forks, stars, url, name } : RepositoryProps) => {
	return (
		<>
			<th><a href={ url }>{ name }</a></th>
			<th>{ stars }</th>
			<th>{ forks }</th>
		</>
	);
}