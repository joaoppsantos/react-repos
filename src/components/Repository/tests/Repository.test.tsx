import * as React from 'react';
import { render } from '@testing-library/react';
import { Repository } from '../Repository';

describe('<Repository />', () => {
    it('should render correctly', () => {
        const mockRepository = {
            forks: 10,
            stars: 15,
            url: 'www.mock-repository.com',
            name: 'React'
        }
        const { container } = render(
            <Repository { ...mockRepository } />
        );

        expect(container).toMatchSnapshot();
    });
});
