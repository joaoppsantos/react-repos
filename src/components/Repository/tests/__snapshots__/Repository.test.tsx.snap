// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> should render correctly 1`] = `
<div>
  <th>
    <a
      href="www.mock-repository.com"
    >
      React
    </a>
  </th>
  <th>
    15
  </th>
  <th>
    10
  </th>
</div>
`;
